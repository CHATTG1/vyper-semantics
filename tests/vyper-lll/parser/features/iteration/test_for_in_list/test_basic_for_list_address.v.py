@public
def data() -> address:
    addresses = [
        0x7d577a597B2742b498Cb5Cf0C26cDCD726d39E6e,
        0x82A978B3f5962A5b0957d9ee9eEf472EE55B42F1,
        0xDCEceAF3fc5C0a63d195d69b1A90011B7B19650D
    ]
    count = 0
    for i in addresses:
        count += 1
        if count == 2:
            return i
    return 0x0000000000000000000000000000000000000000
