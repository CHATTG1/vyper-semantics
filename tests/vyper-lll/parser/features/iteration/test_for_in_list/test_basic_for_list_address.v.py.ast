%pgm(, ,
  %fdecl(%@public, data, , %address,
    %assign(%var(addresses), %list(%hex("7d577a597B2742b498Cb5Cf0C26cDCD726d39E6e") %hex("82A978B3f5962A5b0957d9ee9eEf472EE55B42F1") %hex("DCEceAF3fc5C0a63d195d69b1A90011B7B19650D")))
    %assign(%var(count), 0)
    %forlist(i, %var(addresses),
      %augassign(+=, %var(count), 1)
      %if(%compareop(%eq, %var(count), 2),
        %return(%var(i))))
    %return(%hex("0000000000000000000000000000000000000000")))
)
