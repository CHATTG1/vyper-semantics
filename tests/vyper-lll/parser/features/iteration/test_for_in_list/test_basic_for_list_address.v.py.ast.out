$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1943314746 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $with ( $_L , 320 , $%seq ( $mstore ( $add ( $mul ( 32 , $uclamplt ( 0 , 3 ) ) , $_L ) , 715574669332965331462488905126228088406116900462 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 1 , 3 ) ) , $_L ) , 745948140856946866108753121277737810491401257713 ) , $mstore ( $add ( $mul ( 32 , $uclamplt ( 2 , 3 ) ) , $_L ) , 1261261412747064473059787654210242173679882036493 ) ) ) , $mstore ( 416 , 0 ) , $repeat ( 480 , 0 , 3 , $%seq ( $mstore ( 448 , $mload ( $add ( 320 , $mul ( $mload ( 480 ) , 32 ) ) ) ) , $%seq ( $with ( $_mloc , 416 , $mstore ( $_mloc , $clamp ( $mload ( 96 ) , $add ( $mload ( $_mloc ) , 1 ) , $mload ( 64 ) ) ) ) , $if ( $eq ( $mload ( 416 ) , 2 ) , $%seq ( $%seq ( $mstore ( 0 , $mload ( 448 ) ) , $return ( 0 , 32 ) ) ) ) ) ) ) , $%seq ( $mstore ( 0 , 0 ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
